build.py

```diff
@@ -213,6 +213,9 @@ def do_make(target=None, log=False):
     if not git_describe:
         git_describe = subprocess.check_output('git describe', shell=True).strip()
 
+if os.environ.has_key('BRANCH'):
```

Comment:
While at it, could you clean up the git_describe override and add it here (instead of above)?  Then you can remove the 'if not git_describe' part in the previous check.